src/applications/bmqtool/m_bmqtool_inpututil.cpp

```diff
@@ -319,5 +501,57 @@ bool InputUtil::populateSubscriptions(bmqt::QueueOptions*              out,
     return !failed;
 }
 
+bool InputUtil::decodeHexDump(bsl::ostream& out,
+                              bsl::istream& in,
+                              bsl::string*  error)
+{
+    char outputBuffer[4];  // should be equal to k_BLOCK_SIZE in
+                           // bdlb::Print::hexDump()
+    bdlde::HexDecoder hexDecoder;
+
+    bsl::string line;
+    int         numOut = 0;
+    int         numIn  = 0;
+
+    while (bsl::getline(in, line)) {
+        if (line.empty())
+            continue;  // skip empty lines
+
+        // Sanity check
+        if (line.find(' ') == bsl::string::npos) {
+            if (error)
+                error->assign(
+                    "Wrong hexdump format, space delimeter is not detected");
+            return false;
+        }
+
+        // Convert hexdump to binary, see format in bdlb::Print::hexDump()
+        bdlb::Tokenizer           tokenizer(line, " ");
+        bdlb::Tokenizer::iterator tokenizerIt = tokenizer.begin();
+        // Process tokens skipping first one (address offset)
+        for (++tokenizerIt; tokenizerIt != tokenizer.end(); ++tokenizerIt) {
+            bslstl::StringRef token = *tokenizerIt;
+
+            // Stop when ASCII representation is detected
+            if (token.at(0) == '|')
+                break;
+
+            int status = hexDecoder.convert(outputBuffer,
```

Comment:
```suggestion
            const int status = hexDecoder.convert(outputBuffer,
```

Might also change the name to the expected `rc`